templates/terraform/constants/secret_version.go.erb

```diff
@@ -1,3 +1,4 @@
+<% unless compiler == "terraformobjectlibrary-codegen" -%>
```

Comment:
I'm confused why removing this piece of code is important to conversion.

Reply:
@rileykarson it is important to remove this code because it is specific to the terraform-provider-google generated code. 

Code that has signatures like 

- resource{service.resource}Update(d *schema.ResourceData, meta interface{})
- resource{service.resource}Read(d, meta)

only work with the code generated for terraform-provider-google.

the call to `resourceSecretManagerSecretVersionRead(d, meta)` gives a compilation error because `resourceSecretManagerSecretVersionRead ` does not exist in the conversion code.

This is a case where the shared template in the constant folder is inserting a terraform-provider-google only code.

In other cases, the specific provider code was inserted by a handwritten resource like in this example

https://github.com/GoogleCloudPlatform/magic-modules/blob/d910e58b30b67e1c641463ab56de084ed4351225/third_party/terraform/resources/resource_logging_billing_account_sink.go#L67

Reply:
Hmm- can you add a stub method with that signature in conversion using a handwritten file? I'd rather we take on complexity that way than doing so in the direct provider code / templates.

Reply:
I will try adding a stub for it

Reply:
keeping the conditional based in the precedent